### PR TITLE
add/update permalinks to not use .html

### PR DESCRIPTION
**Pull request by therewasaguy**
- remove .html extension from events and posts permalinks
- add permalinks for header nav links

### CODE REVIEW

**Review by khoeger — COMMENTED** [review]
The main pages no longer have .html, but specific events, such as February 2017's Music Visualization event or the upcoming Hip Hop Hacks 2017, still end in .html.

**Comment by therewasaguy** [comment]
@khoeger the files still have .html extensions, but the links that we use shouldn't. Let me know if you see a spot where a link includes ".html".

Both of these links already work, and this PR would just change the permalink so that, throughout the site, we only link to the first one
https://musichackathon.github.io/mmh_jekyll/2017-06-10
https://musichackathon.github.io/mmh_jekyll/2017-06-10.html

**Comment by khoeger** [comment]
I checked out the branch permalinks, but still got http://localhost:4000/2017-06-10.html. [](url)

**Comment by therewasaguy** [comment]
Ah, thanks @khoeger good catch. Just updated to use what Jekyll calls "pretty urls" for events and posts. This makes a folder with index.html inside of it. This changes relative paths so I updated `_layouts/event.html` to use absolute path for images

File: _layouts/event.html [hunk 1]
```diff
@@ -8,7 +8,7 @@ <h1 class="post-title">{{ page.title | escape }}</h1>
   </header>
 
   {% if page.image %}
-    <img src="assets/events/{{page.image}}"/>
+    <img src="/assets/events/{{page.image}}"/>
```
Comment (trevorknight) [inline _layouts/event.html]: This will break when hosted by github. 

Please use {{ site.github.url }}/assets

File: blog.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Blog
 layout: blog
+permalink: blog
```
Comment (trevorknight) [inline blog.md]: I know this is my fault, not yours, but as long as you're editing this front matter, would be willing to make the pages have the fields in the same order?

**Review by trevorknight — CHANGES_REQUESTED** [review]

**Review by trevorknight — APPROVED** [review]

**Pull request merged**